File: system-test/java/src/main/java/com/mycompany/myshop/testkit/driver/port/dtos/BrowseCouponsResponse.java
package com.mycompany.myshop.testkit.driver.port.dtos;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Instant;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class BrowseCouponsResponse {
    private List<CouponDto> coupons;

    @Data
    @Builder
    @NoArgsConstructor
    @AllArgsConstructor
    public static class CouponDto {
        private String code;
        private double discountRate;
        private Instant validFrom;
        private Instant validTo;
        private Integer usageLimit;
        private Integer usedCount;
    }
}

File: system-test/typescript/src/testkit/driver/port/dtos/BrowseCouponsResponse.ts
export interface BrowseCouponItem {
  code: string;
  discountRate: number;
  validFrom?: string;
  validTo?: string;
  usageLimit?: number;
  usedCount: number;
}

export interface BrowseCouponsResponse {
  coupons: BrowseCouponItem[];
}
